azure/scope/managedmachinepool.go

```diff
@@ -219,6 +219,16 @@ func (s *ManagedMachinePoolScope) DeleteLongRunningOperationState(name, service
 	futures.Delete(s.ControlPlane, name, service)
 }
 
+// UpdateMachinePoolReplicas updates capi machinepool replica count.
+func (s *ManagedMachinePoolScope) UpdateMachinePoolReplicas(replicas *int32) error {
+	patch := client.MergeFrom(s.MachinePool.DeepCopy())
+	s.MachinePool.Spec.Replicas = replicas
+	if err := s.Client.Patch(context.Background(), s.MachinePool, patch); err != nil {
```

Comment:
we shouldn't use `context.Background()` here, we have an actual context we can pass in from the parent func

Reply:
I assume you mean that the calling func (`func (s *Service) Reconcile`) has its own context object, and we should pass it along to `UpdateMachinePoolReplicas` here. I agree.

So we'd update the interface accordingly:

`UpdateMachinePoolReplicas(ctx context.Context, replicas *int32) error`

And then the actual method definition and its invocation in the flow of `Reconcile()` w/ the updated signature.